### PR TITLE
Explicit versioning for pip requirements based on datasource type

### LINKED ISSUE #15427: pip requirements with git references create the update incorrectly
### How are you running Renovate?

WhiteSource Renovate hosted app on github.com

### If you're self-hosting Renovate, tell us what version of Renovate you run.

_No response_

### Please select which platform you are using if self-hosting.

_No response_

### If you're self-hosting Renovate, tell us what version of the platform you run.

_No response_

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the bug

`==` is being added to the newValue when creating an update. While this is valid for pip packages install from repositories, it isn't in the case of git references.

Sample repo with problematic PR: https://github.com/mikhailadvani/use_pip_package/pull/2

### Relevant debug logs

<details><summary>Logs</summary>

https://github.com/mikhailadvani/use_pip_package/runs/6276936064?check_suite_focus=true
</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction repository in the bug description

### PR DESCRIPTION
**Pull request by mikhailadvani**
## Changes

Set versioning type for pip requirements based on repository type.

## Context

Closes #15427

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/pip_requirements/extract.ts [hunk 1]
```diff
@@ -111,6 +114,7 @@ export function extractPackageFile(
         depName,
         currentValue,
         datasource: PypiDatasource.id,
+        versioning: pep440Versioning.id,
```
Comment (rarkins) [inline lib/modules/manager/pip_requirements/extract.ts:117]: No need to set this explicitly as pypi datasource defaults to it 
```suggestion
```

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/modules/manager/pip_requirements/extract.ts (hunk 1 above)
Comment (mikhailadvani) [reply lib/modules/manager/pip_requirements/extract.ts]: fixed

**Review by viceice — COMMENTED** [review]
i would not set versioning at all and let datasource use default versioning.so it's more easy to let users override as required.

**Comment by mikhailadvani** [comment]
Overrides would still be available right? This is to keep the default behaviour sane. In the current scenario, the PR raised corrupts the requirements.txt. If it had skipped raising the PR completely and it would have been configurable using regex and package rules alone, I would be more inclined to retain default versioning. Your thoughts @viceice ?

**Comment by mikhailadvani** [comment]
Sample problematic PR which is mentioned in the issue as well: https://github.com/mikhailadvani/use_pip_package/pull/2

**Comment by rarkins** [comment]
git tags defaults to semver versioning today, doesn't it? So I don't understand why/where pep440 versioning is being used for a git tags reference

**Comment by viceice** [comment]
this is not the right solution, you need to debug why pep440 is used instead of datasource semver default for git-tags

**Comment by mikhailadvani** [comment]
The challenge is that it was working fine when 32.34.1 was released. there is some runtime component which has changed this behaviour. unfortunately, I am not able to identify the root cause. based on my theory that the behaviour is modifiable at runtime, I think it to be a more robust solution to be explicit with the versioning scheme. 

whether the versioning scheme should be semver or loose is definitely up for debate but given people do use YYMMDD versioning, I chose to go in for loose. A popular repo that does this versioning - https://github.com/awslabs/amazon-eks-ami/tags

**Review by viceice — CHANGES_REQUESTED** [review]
nope, this is a workaround for a maabe bug, not a solution we like to use

**Pull request closed without merging**